parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -126,22 +122,18 @@ wait_for_file 3600 1 {{GetCustomSearchDomainsCSEScriptFilepath}} || exit $ERR_FI
 {{GetCustomSearchDomainsCSEScriptFilepath}} > /opt/azure/containers/setup-custom-search-domain.log 2>&1 || exit $ERR_CUSTOM_SEARCH_DOMAINS_FAIL
 {{end}}
 
-{{- if IsDockerContainerRuntime}}
-ensureDocker
-{{else if IsKataContainerRuntime}}
+{{- if IsKataContainerRuntime}}
 if grep -q vmx /proc/cpuinfo; then
     installKataContainersRuntime
-fi
+fi 
 {{end}}
 
+ensureContainerRuntime
+
 configureK8s
 
 configureCNI
 
-{{- if NeedsContainerd}}
```

Comment:
If possible I would still prefer this if-else check in template, so that the generated template is always just for one container runtime, and it will save some bytes on the cloud init data

Reply:
Will change all containerruntime checks to use template if vars